### PR TITLE
Added development guide.

### PR DESCRIPTION
**Pull request by jorgecarleitao**
Describing how I have been merging PRs. Also removes some of the items from the README.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+2. Label the PR accordingly (`Testing`, `Documentation`, `Enchancement`, `Feature`, `Bug`)
+3. If the PR is backward incompatible:
+    1. create a new issue labeled `backward-incompatible` with what changed and how to migrate
+       from the old API to the new API
+    2. Edit the PR's description with `Closes #...`
+4. Adjust the PRs title with a description suitable for the changelog
+    * In the past tense
+    * backtick code names (e.g. `MutableArray`, not MutableArray)
+    * place "(-X%)" or "(Yx)" at the end if it is a performance improvement
+5. Press merge and remove any items added by github from the description
```
Comment (houqp) [inline DEVELOPMENT.md:37]: If a contributor adds multiple commits in a PR each with different commit messages, some useful, some not. What's the expected behavior? Should we manually clean up the generated commit message and keep only the useful ones?

Comment (ritchie46) [reply DEVELOPMENT.md:37]: > What's the expected behavior? Should we manually clean up the generated commit message and keep only the useful ones?

In that regard, maybe a squash merge is less maintenance?

Comment (jorgecarleitao) [reply DEVELOPMENT.md:37]: What about

> Press merge (squash merge) and adjust any items added by github to your liking

Comment (houqp) [reply DEVELOPMENT.md:37]: That sounds good :+1:

**Pull request merged**